crates/daphne/src/pine/vdaf.rs

```diff
@@ -452,38 +463,43 @@ impl<F: FftFriendlyFieldElement, X: Xof<SEED_SIZE>, const SEED_SIZE: usize>
             let corrected_vf_joint_rands = {
                 X::seed_stream(
                     &corrected_vf_joint_rand_seed,
-                    &self.flp.cfg.dst(USAGE_VF_JOINT_RAND),
-                    &[self.flp.cfg.param.num_proofs],
-                )
-                .into_field_vec(
-                    self.flp.joint_rand_len() * usize::from(self.flp.cfg.param.num_proofs),
+                    &dst(self.algorithm_id, USAGE_VF_JOINT_RAND),
+                    &[self.num_proofs],
                 )
+                .into_field_vec(self.flp.joint_rand_len() * usize::from(self.num_proofs))
             };
 
             let query_rands = {
-                let mut xof = X::init(verify_key, &self.flp.cfg.dst(USAGE_QUERY_RAND));
-                xof.update(&[self.flp.cfg.param.num_proofs]);
+                let mut xof = X::init(verify_key, &dst(self.algorithm_id, USAGE_QUERY_RAND));
+                xof.update(&[self.num_proofs_sq_norm_equal, self.num_proofs]);
                 xof.update(nonce);
                 xof.into_seed_stream().into_field_vec(
-                    self.flp_sq_norm_equal.query_rand_len()
-                        + self.flp.query_rand_len() * usize::from(self.flp.cfg.param.num_proofs),
+                    self.flp_sq_norm_equal.query_rand_len() * num_proofs_sq_norm_equal
+                        + self.flp.query_rand_len() * num_proofs,
                 )
             };
-            let (qr_sq_norm_equal, qr) =
-                query_rands.split_at(self.flp_sq_norm_equal.query_rand_len());
+            let (qr_sq_norm_equal, qr) = query_rands
+                .split_at(self.flp_sq_norm_equal.query_rand_len() * num_proofs_sq_norm_equal);
 
             let mut verifiers_share = Vec::with_capacity(
                 self.flp_sq_norm_equal.verifier_len()
-                    + self.flp.verifier_len() * usize::from(self.flp.cfg.param.num_proofs),
+                    + num_proofs_sq_norm_equal
+                    + self.flp.verifier_len() * num_proofs,
             );
 
-            verifiers_share.append(&mut self.flp_sq_norm_equal.query(
-                &meas_share,
-                ps_sq_norm_equal,
-                qr_sq_norm_equal,
-                &[],
-                2,
-            )?);
+            for (proof_share, query_rand) in ps_sq_norm_equal
+                .chunks(self.flp_sq_norm_equal.proof_len())
```

Comment:
if the vector splits evenly, it's better to use `chunks_exact`

Reply:
Nice, that's possible basically for every call to `chunks()` in this file. I've tacked on a commit.